ehrapy/anndata/anndata_ext.py

```diff
@@ -154,63 +154,130 @@ def anndata_to_df(
+        adata._inplace_subset_var(~indices)
+        adata.obs = adata.obs.join(df)
+        updated_num_uns, updated_non_num_uns, num_var = _update_uns(adata, to_obs)
+        # cast numerical values from object
+        adata.obs[num_var] = adata.obs[num_var].apply(pd.to_numeric, errors="ignore", downcast="float")
+        # cast non numerical values from object to either bool (if possible) or category
+        adata.obs = _cast_obs_columns(adata.obs)
+        adata.uns["numerical_columns"] = updated_num_uns
+        adata.uns["non_numerical_columns"] = updated_non_num_uns
+
+    return adata
+
+
+def delete_from_obs(adata: AnnData, to_delete: list[str]) -> AnnData:
+    """Delete features from obs.
+
+    Args:
+        adata: The AnnData object
+        to_delete: The columns to delete from obs
+
+    Returns:
+        The original AnnData object with deleted columns from obs.
+    """
+
+    if isinstance(to_delete, str):  # pragma: no cover
+        to_delete = [to_delete]
 
+    if not all(elem in adata.obs.columns.values for elem in to_delete):
+        raise ValueError(f"Columns `{[x for x in to_delete if x not in adata.obs.columns.values]}` are not in obs.")
 
-def move_to_x(adata: AnnData, to_x: list[str] | str) -> AnnData:
+    adata.obs = adata.obs[adata.obs.columns[~adata.obs.columns.isin(to_delete)]]
+
+    return adata
+
+
+def move_to_x(adata: AnnData, to_x: list[str] | str, copy: bool = False) -> AnnData:
     """Move features from obs to X inplace.
 
     Args:
         adata: The AnnData object
         to_x: The columns to move to X
+        copy: Whether to return a copy or not
 
     Returns:
         A new AnnData object with moved columns from obs to X. This should not be used for datetime columns currently.
     """
+
+    if not all(elem in adata.obs.columns.values for elem in to_x):
+        raise ValueError(f"Columns `{[x for x in to_x if x not in adata.obs.columns.values]}` are not in obs.")
+
     if isinstance(to_x, str):  # pragma: no cover
         to_x = [to_x]
-    new_adata = concat([adata, AnnData(adata.obs[to_x], dtype="object")], axis=1)
-    new_adata.obs = adata.obs[adata.obs.columns[~adata.obs.columns.isin(to_x)]]
-    # update uns (copy maybe: could be a costly operation but reduces reference cycles)
-    # users might save those as separate AnnData object and this could be unexpected behaviour if we dont copy
-    num_columns_moved, non_num_columns_moved, _ = _update_uns(adata, to_x, True)
-    new_adata.uns["numerical_columns"] = adata.uns["numerical_columns"] + num_columns_moved
-    new_adata.uns["non_numerical_columns"] = adata.uns["non_numerical_columns"] + non_num_columns_moved
+
+    cols_present_in_x = []
+    cols_not_in_x = []
+
+    for col in to_x:
+        if col in set(adata.var_names):
+            cols_present_in_x.append(col)
+        else:
+            cols_not_in_x.append(col)
+
+    if cols_present_in_x:  # if there are some columns that were already in X, we skip the moving
```

Comment:
The comment should not explain the "what", but the "why". It might not even be necessary at all.